### DISCUSSION TITLE
How to configure queries only from the time ElastAlert is initialized

### DISCUSSION THREAD

**Discussion by ngms17** [jertel/elastalert2#817] [Q&A] [answered]
I noticed that when i run the program, it starts to querie from the time it was shutdown from the last time. How can i define ElastAlert to ignore the past queries and start to run only when i initiate the program?

Thank you

**Comment by jertel** [comment, marked as answer]
https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=scan_entire_timeframe#scan-entire-timeframe

**Reply by ngms17** [reply]
Can´t use this option on a "Any" rule type

**Reply by jertel** [reply]
> Can´t use this option on a "Any" rule type

What makes you think that an "any" rule type cannot use the `scan_entire_timeframe` option?

**Reply by ngms17** [reply]
I have set "scan_entire_timeframe" to "false". I started Elastalert at 13:13 WEST but it still queried from 12:52 WEST to 13:13 WEST:

```
INFO:elastalert:Queried rule Critical Activity Detected from 2022-04-20 12:52 WEST to 2022-04-20 12:57 WEST: 4 / 4 hits
INFO:elastalert:Queried rule Critical Activity Detected from 2022-04-20 12:57 WEST to 2022-04-20 13:02 WEST: 2 / 2 hits
INFO:elastalert:Queried rule Critical Activity Detected from 2022-04-20 13:02 WEST to 2022-04-20 13:07 WEST: 0 / 0 hits
INFO:elastalert:Queried rule Critical Activity Detected from 2022-04-20 13:07 WEST to 2022-04-20 13:12 WEST: 1 / 1 hits
INFO:elastalert:Queried rule Critical Activity Detected from 2022-04-20 13:12 WEST to 2022-04-20 13:13 WEST: 0 / 0 hits
```

And i still received an email alert from an event that occured at 12:53 WEST

**Reply by jertel** [reply]
I misread your initial post. I thought you were asking how to make the query ignore the last run time and always scan the full time frame. That's a common question, so I assumed that's what you were asking. Sorry about that.

You are asking how to force ElastAlert2 to always start the time frame at the time the program is launched. You can use the `--start` command line arg. 

From the [docs](https://elastalert2.readthedocs.io/en/latest/running_elastalert.html?highlight=start#configuration-flags):

![image](https://user-images.githubusercontent.com/16639611/164232291-00fb512f-acc3-4595-b9cf-46ce2d351fd1.png)

**Reply by ngms17** [reply]
Thanks you very much!

Also, is there any way that we can configure ElastAlert2 to run as a service (systemctl)

**Reply by jertel** [reply]
If you're using Docker then you can tell Docker to autostart that container upon server start: https://docs.docker.com/config/containers/start-containers-automatically/#use-a-restart-policy

If you're using Python directly from the command line, yes you can make it run as a service. There are lot of variables in play depending on your Linux distribution so I recommend starting with a Google search for this topic. Or, if you'd like to keep it simple, add this line to your user crontab:

```
@reboot elastalert --arg1 --arg2 --etc &> /some/log/file.log &
```
